Fix dangling reference

dmaslenko opened the pull request:
Fixes compilation error: dangling reference #10134

## Screenshots
N/A

## Testing strategy
The build was tested locally by the commands:
```shell
cd build
cmake -DWITH_XC_ALL=ON -DCMAKE_BUILD_TYPE=Debug -DWITH_TESTS=ON -DWITH_GUI_TESTS=ON ..
make -j8
```

## Type of change
- ✅ Bug fix (non-breaking change that fixes an issue)

dmaslenko committed:
fix compilation error: dangling reference

[What]
  Refactor the Remove.cpp to use a local variable for
  positionalArguments() return.

[Why]
  To fix the compilation issue:
    error: possibly dangling reference to a temporary

  The parser->positionalArguments() function returns a
  temporary QList object, and at(1) returns a reference
  to an element in that list.

  But the temporary QList object is destroyed after this
  line of code, leaving entryPath to be a dangling reference.

dmaslenko committed:
fix compilation warning: remove dead semicolon

droidmonkey commented on src/cli/Remove.cpp:
```diff
@@ -35,7 +35,8 @@ int Remove::executeWithDatabase(QSharedPointer<Database> database, QSharedPointe
 
-    auto& entryPath = parser->positionalArguments().at(1);
+    auto args = parser->positionalArguments();
+    auto& entryPath = args.at(1);
```
Even easier, just remove the & after auto in the original line. I don't think this is actually an error though, seems like the compiler is getting too sensitive.

dmaslenko replied:
Thank you for suggestion, fixed. I had the same idea before but checked the code and found some examples with a separate variable.

dmaslenko committed:
Simplify entryPath assignment in Remove.cpp

The previous code assigned entryPath in two steps, first by getting the
positional arguments from the parser, and then by selecting the second argument.
This was modified to a single-line assignment for better readability and code simplicity.
